src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3466,9 +3466,20 @@ private static bool IsValidExpressionBody(SyntaxNode expressionSyntax, BoundExpr
         /// <summary>
         /// Binds an expression-bodied member with expression e as either { return e; } or { e; }.
         /// </summary>
-        internal virtual BoundBlock BindExpressionBodyAsBlock(ArrowExpressionClauseSyntax expressionBody,
-                                                      BindingDiagnosticBag diagnostics)
+        internal virtual BoundBlock BindExpressionBodyAsBlock(
+            ArrowExpressionClauseSyntax expressionBody,
+            BindingDiagnosticBag diagnostics)
         {
+            var messageId = expressionBody.Parent switch
+            {
+                ConstructorDeclarationSyntax or DestructorDeclarationSyntax => MessageID.IDS_FeatureExpressionBodiedDeOrConstructor,
+                AccessorDeclarationSyntax => MessageID.IDS_FeatureExpressionBodiedAccessor,
+                BaseMethodDeclarationSyntax => MessageID.IDS_FeatureExpressionBodiedMethod,
+                _ => (MessageID?)null,
```

Comment:
Is this a scenario we expect to see? Should we just throw?

Reply:
Yes.  Local functions.  Note: i'm happy to explicitly check for those and throw in the rest if you prefer that.

Reply:
Yes please. If any new cases come through here, they will likely need to be handled in some fashion.

Reply:
sounds good.  eill do!

Reply:
this actually made thigns *much* cleaner.  It allowed the existing binding checks to all be unified in this single location :)